--- ./cmake/ssl.cmake
+++ ,/cmake/ssl.cmake
@@ -139,8 +139,10 @@ MACRO (MYSQL_CHECK_SSL)
     ENDIF()
 
     # On mac this list is <.dylib;.so;.a>
-    # We prefer static libraries, so we revert it here.
-    LIST(REVERSE CMAKE_FIND_LIBRARY_SUFFIXES)
+    # If we prefer static libraries, we revert it here.
+    IF (NOT WITH_SHARED_OPENSSL)
+        LIST(REVERSE CMAKE_FIND_LIBRARY_SUFFIXES)
+    ENDIF()
     MESSAGE(STATUS "suffixes <${CMAKE_FIND_LIBRARY_SUFFIXES}>")
     FIND_LIBRARY(OPENSSL_LIBRARIES
                  NAMES ssl ssleay32 ssleay32MD
@@ -148,7 +150,9 @@ MACRO (MYSQL_CHECK_SSL)
     FIND_LIBRARY(CRYPTO_LIBRARY
                  NAMES crypto libeay32
                  HINTS ${OPENSSL_ROOT_DIR}/lib)
-    LIST(REVERSE CMAKE_FIND_LIBRARY_SUFFIXES)
+    IF (NOT WITH_SHARED_OPENSSL)
+        LIST(REVERSE CMAKE_FIND_LIBRARY_SUFFIXES)
+    ENDIF()
 
     # Verify version number. Version information looks like:
     #   #define OPENSSL_VERSION_NUMBER 0x1000103fL
