--- ./src/google/protobuf/testing/googletest.cc
+++ ./src/google/protobuf/testing/googletest.cc
@@ -104,6 +104,7 @@ string GetTemporaryDirectoryName() {
   if (HasPrefixString(result, "\\")) {
     result.erase(0, 1);
   }
+  std::replace(result.begin(), result.end(), '\\', '/');
 #endif  // _WIN32
   return result;
 }
--- ./src/google/protobuf/testing/googletest.cc
+++ ./src/google/protobuf/testing/googletest.cc
@@ -48,6 +48,8 @@
 #include <fcntl.h>
 #include <iostream>
 #include <fstream>
+#include <algorithm>
+#include <string>
 
 namespace google {
 namespace protobuf {
