--- ./cpp/src/IceUtil/Exception.cpp
+++ ./cpp/src/IceUtil/Exception.cpp
@@ -36,20 +36,6 @@
 #  define ICE_GCC_STACK_TRACES
 #endif
 
-#if defined(_WIN32) && !defined(ICE_OS_WINRT) && !defined(__MINGW32__)
-#  define ICE_WIN32_STACK_TRACES
-#  if defined(_MSC_VER) && _MSC_VER >= 1700
-#    define DBGHELP_TRANSLATE_TCHAR
-#    include <IceUtil/StringConverter.h>
-#    if _MSC_VER >= 1900
-#       // VS 2015 RC issues this warning for code in DbgHelp.h
-#       pragma warning(disable:4091)
-#    endif
-#  endif
-#  include <DbgHelp.h>
-#  include <tchar.h>
-#endif
-
 using namespace std;
 
 namespace IceUtilInternal
