--- ./qtbase/qmake/generators/unix/unixmake2.cpp
+++  ./qtbase/qmake/generators/unix/unixmake2.cpp
@@ -788,8 +788,9 @@ UnixMakefileGenerator::writeMakeParts(QT
                 break;
             }
             bool isApp = (project->first("TEMPLATE") == "app");
+            bool isFramework = !isApp && project->isEmpty("QMAKE_BUNDLE_EXTENSION");
             QString info_plist_out = escapeFilePath(
-                    bundle_dir + (isApp ? "Contents/Info.plist"
+                    bundle_dir + (!isFramework ? "Contents/Info.plist"
                                         : "Versions/" + project->first("QMAKE_FRAMEWORK_VERSION")
                                           + "/Resources/Info.plist"));
             bundledFiles << info_plist_out;
