--- ./qtbase/src/corelib/global/qprocessordetection.h
+++ ./qtbase/src/corelib/global/qprocessordetection.h
@@ -337,6 +337,23 @@
 #  ifdef __SIZEOF_POINTER__
      /* GCC & friends define this */
 #    define Q_PROCESSOR_WORDSIZE        __SIZEOF_POINTER__
+#  elif defined(Q_PROCESSOR_X86_32)
+     /* GCC 4.1 on CentOS 5.10 doesn't define __SIZEOF_POINTER__,
+      * so without this special case, we would hit the case where
+      * Q_PROCESSOR_WORDSIZE get set to QT_POINTER_SIZE.
+      *
+      * When that's the case, we can't run "./configure", because we can't
+      * successfully bootstrap qmake without a qconfig.h which defines the
+      * correct Q_POINTER_SIZE. However, bootstrapping qmake happens before
+      * configure generates a proper qconfig.h for the architecture, so we
+      * have a chicken-and-egg problem.
+      *
+      * To fix our build issue, we simply define Q_PROCESSOR_WORDSIZE to 4
+      * when we know we're on a 32-bit x86 system.
+      * If we ever have to build on CentOS 5.10 with another architecture,
+      * building qmake would fail as described above, and a special case
+      * similar to this can be added for that architecture. */
+#    define Q_PROCESSOR_WORDSIZE        4
 #  elif defined(_LP64) || defined(__LP64__) || defined(WIN64) || defined(_WIN64)
 #    define Q_PROCESSOR_WORDSIZE        8
 #  else
