--- ./qtbase/src/plugins/bearer/bearer.pro
+++ ./qtbase/src/plugins/bearer/bearer.pro
@@ -10,7 +10,6 @@ win32:SUBDIRS += generic
 blackberry:SUBDIRS += blackberry
 win32:!wince*:SUBDIRS += nativewifi
 mac:contains(QT_CONFIG, corewlan):SUBDIRS += corewlan
-mac:SUBDIRS += generic
 android:!android-no-sdk:SUBDIRS += android
 
-isEmpty(SUBDIRS):SUBDIRS = generic
+isEmpty(SUBDIRS):error("No no bearer plugins enabled.")
