--- ./qtbase/qmake/generators/win32/msvc_nmake.cpp
+++ ./qtbase/qmake/generators/win32/msvc_nmake.cpp
@@ -531,13 +531,13 @@
             if (objDir == ".\\")
                 objDir = "";
             for(QStringList::Iterator cppit = Option::cpp_ext.begin(); cppit != Option::cpp_ext.end(); ++cppit)
-                t << '{' << escapeDependencyPath(sourceDir) << '}' << (*cppit)
-                  << '{' << escapeDependencyPath(objDir) << '}' << Option::obj_ext << "::\n\t"
+                t << '{' << sourceDir << '}' << (*cppit)
+                  << '{' << objDir << '}' << Option::obj_ext << "::\n\t"
                   << var("QMAKE_RUN_CXX_IMP_BATCH").replace(QRegExp("\\$@"), fileVar("OBJECTS_DIR"))
                   << "\n\t$<\n<<\n\n";
             for(QStringList::Iterator cit = Option::c_ext.begin(); cit != Option::c_ext.end(); ++cit)
-                t << '{' << escapeDependencyPath(sourceDir) << '}' << (*cit)
-                  << '{' << escapeDependencyPath(objDir) << '}' << Option::obj_ext << "::\n\t"
+                t << '{' << sourceDir << '}' << (*cit)
+                  << '{' << objDir << '}' << Option::obj_ext << "::\n\t"
                   << var("QMAKE_RUN_CC_IMP_BATCH").replace(QRegExp("\\$@"), fileVar("OBJECTS_DIR"))
                   << "\n\t$<\n<<\n\n";
         }
