2017-01-21:
	Added -Zc:threadSafeInit- to CFLAGS to work around QTBUG-58318

--- ./qtbase/mkspecs/common/msvc-desktop.conf
+++ ./qtbase/mkspecs/common/msvc-desktop.conf
@@ -25,11 +25,11 @@ QMAKE_LEX               = flex
 QMAKE_LEXFLAGS          =
 QMAKE_YACC              = byacc
 QMAKE_YACCFLAGS         = -d
-QMAKE_CFLAGS            = -nologo -Zc:wchar_t
+QMAKE_CFLAGS            = -nologo -Zc:wchar_t -Zi -Zc:threadSafeInit-
 QMAKE_CFLAGS_WARN_ON    = -W3
 QMAKE_CFLAGS_WARN_OFF   = -W0
-QMAKE_CFLAGS_RELEASE    = -O2 -MD
-QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += -O2 -MD -Zi
+QMAKE_CFLAGS_RELEASE    = -Ox -GF -MD
+QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += -Ox -GF -MD -Zi
 QMAKE_CFLAGS_DEBUG      = -Zi -MDd
 QMAKE_CFLAGS_YACC       =
 QMAKE_CFLAGS_LTCG       = -GL
@@ -65,14 +65,14 @@ QMAKE_RUN_CXX_IMP       = $(CXX) -c $(CX
 QMAKE_RUN_CXX_IMP_BATCH = $(CXX) -c $(CXXFLAGS) $(INCPATH) -Fo$@ @<<
 
 QMAKE_LINK              = link
-QMAKE_LFLAGS            = /NOLOGO /DYNAMICBASE /NXCOMPAT
+QMAKE_LFLAGS            = /DEBUG /OPT:REF /OPT:ICF /NOLOGO /DYNAMICBASE /NXCOMPAT
 QMAKE_LFLAGS_RELEASE    = /INCREMENTAL:NO
 QMAKE_LFLAGS_RELEASE_WITH_DEBUGINFO = /DEBUG /OPT:REF /INCREMENTAL:NO
 QMAKE_LFLAGS_DEBUG      = /DEBUG
 QMAKE_LFLAGS_CONSOLE    = /SUBSYSTEM:CONSOLE
 QMAKE_LFLAGS_WINDOWS    = /SUBSYSTEM:WINDOWS
 QMAKE_LFLAGS_EXE        = \"/MANIFESTDEPENDENCY:type=\'win32\' name=\'Microsoft.Windows.Common-Controls\' version=\'6.0.0.0\' publicKeyToken=\'6595b64144ccf1df\' language=\'*\' processorArchitecture=\'*\'\"
-QMAKE_LFLAGS_DLL        = /DLL
+QMAKE_LFLAGS_DLL        = /DEBUG /DLL
 QMAKE_LFLAGS_LTCG       = /LTCG
 QMAKE_EXTENSION_STATICLIB = lib
 
