--- ./cmake/install_layout.cmake
+++ ./cmake/install_layout.cmake
@@ -145,14 +145,8 @@ SET(INSTALL_PLUGINTESTDIR_STANDALONE
 SET(INSTALL_BINDIR_RPM                  "bin")
 SET(INSTALL_SBINDIR_RPM                 "sbin")
 SET(INSTALL_SCRIPTDIR_RPM               "bin")
-#
-IF(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
-  SET(INSTALL_LIBDIR_RPM                "lib64")
-  SET(INSTALL_PLUGINDIR_RPM             "lib64/mysql/plugin")
-ELSE()
-  SET(INSTALL_LIBDIR_RPM                "lib")
-  SET(INSTALL_PLUGINDIR_RPM             "lib/mysql/plugin")
-ENDIF()
+SET(INSTALL_LIBDIR_RPM                  "lib")
+SET(INSTALL_PLUGINDIR_RPM               "lib/mysql/plugin")
 #
 SET(INSTALL_INCLUDEDIR_RPM              "include/mysql")
 #
