Ported from mumble-developers-qt/4.8-mumble
59f1692e5724778f543f2b9708963377b7ba5019

--- ./qtbase/src/network/socket/qlocalserver_win.cpp
+++ ./qtbase/src/network/socket/qlocalserver_win.cpp
@@ -53,7 +53,7 @@
 // The buffer size need to be 0 otherwise data could be
 // lost if the socket that has written data closes the connection
 // before it is read.  Pipewriter is used for write buffering.
-#define BUFSIZE 0
+#define BUFSIZE 4096
 
 // ###: This should be a property. Should replace the insane 50 on unix as well.
 #define SYSTEM_MAX_PENDING_SOCKETS 8
@@ -311,8 +311,10 @@ void QLocalServerPrivate::closeServer()
     connectionEventNotifier->deleteLater();
     connectionEventNotifier = 0;
     CloseHandle(eventHandle);
-    for (int i = 0; i < listeners.size(); ++i)
+    for (int i = 0; i < listeners.size(); ++i) {
+        FlushFileBuffers(listeners[i].handle);
         CloseHandle(listeners[i].handle);
+    }
     listeners.clear();
 }
 
