--- ./qtbase/src/plugins/bearer/networkmanager/qnetworkmanagerservice.h
+++ ./qtbase/src/plugins/bearer/networkmanager/qnetworkmanagerservice.h
@@ -497,7 +497,7 @@ public:
                             QObject *parent = 0)
         : QDBusAbstractInterface(service, path, interface.toLatin1().data(), connection, parent)
     {}
-    ~PropertiesDBusInterface() = default;
+    ~PropertiesDBusInterface() {}
 };
 
 QT_END_NAMESPACE
