--- ./cpp/config/Make.rules.mak
+++ ./cpp/config/Make.rules.mak
@@ -31,7 +31,7 @@ prefix = $(PREFIX)
 
 #
 # Specify your C++ compiler, or leave unset for auto-detection.
-# Supported values are: VC90, VC100, VC110
+# Supported values are: VC90, VC100, VC110, VC120
 #
 # CPP_COMPILER = VCxxx
 
@@ -79,12 +79,16 @@ THIRDPARTY_HOME = $(PROGRAMFILES)\ZeroC\
 #
 !if "$(CPP_COMPILER)" == ""
 
-!if "$(VISUALSTUDIOVERSION)" == "11.0"
+!if "$(VISUALSTUDIOVERSION)" == "12.0"
+CPP_COMPILER            = VC120
+!elseif "$(VISUALSTUDIOVERSION)" == "11.0"
 CPP_COMPILER            = VC110
 !elseif ([cl 2>&1 | findstr "Version\ 16" > nul] == 0)
 CPP_COMPILER            = VC100
 !elseif ([cl 2>&1 | findstr "Version\ 17" > nul] == 0)
 CPP_COMPILER            = VC110
+!elseif ([cl 2>&1 | findstr "Version\ 18" > nul] == 0)
+CPP_COMPILER            = VC120
 !elseif ([cl 2>&1 | findstr "Version\ 15" > nul] == 0)
 CPP_COMPILER            = VC90
 !else
@@ -92,7 +96,7 @@ CPP_COMPILER            = VC90
 !endif
 
 #!message CPP_COMPILER set to $(CPP_COMPILER)
-!elseif "$(CPP_COMPILER)" != "VC90" && "$(CPP_COMPILER)" != "VC100" && "$(CPP_COMPILER)" != "VC110"
+!elseif "$(CPP_COMPILER)" != "VC90" && "$(CPP_COMPILER)" != "VC100" && "$(CPP_COMPILER)" != "VC110" && "$(CPP_COMPILER)" != "VC120"
 !error Invalid CPP_COMPILER setting: $(CPP_COMPILER). Must be one of: VC90, VC100 or VC110.
 !endif
 
@@ -141,6 +145,7 @@ SETARGV			= setargv.obj
 #
 !include        $(top_srcdir)/config/Make.rules.msvc
 
+# fixme(mkrautz): allow VC120
 !if "$(WINRT)" == "yes" && "$(CPP_COMPILER)" != "VC110"
 !error CPP_COMPILER: $(CPP_COMPILER) not supported to build Ice for WinRT
 !endif
@@ -149,6 +154,8 @@ SETARGV			= setargv.obj
 libsuff			= \vc90$(x64suffix)
 !elseif "$(CPP_COMPILER)" == "VC110"
 libsuff			= \vc110$(x64suffix)
+!elseif "$(CPP_COMPILER)" == "VC120"
+libsuff			= \vc120$(x64suffix)
 !else
 libsuff			= $(x64suffix)
 !endif
@@ -168,6 +175,8 @@ COMPSUFFIX	= _vc90
 COMPSUFFIX	= _vc100
 !elseif "$(CPP_COMPILER)" == "VC110"
 COMPSUFFIX  = _vc110
+!elseif "$(CPP_COMPILER)" == "VC120"
+COMPSUFFIX  = _vc120
 !endif
 !endif
 
