--- ./glib/gmessages.c
+++ ./glib/gmessages.c
@@ -1842,7 +1842,7 @@ g_log_writer_supports_color (gint output
   return isatty (output_fd);
 }
 
-#ifdef __linux__
+#if 0
 static int journal_fd = -1;
 
 #ifndef SOCK_CLOEXEC
@@ -1881,7 +1881,7 @@ open_journal (void)
 gboolean
 g_log_writer_is_journald (gint output_fd)
 {
-#ifdef __linux__
+#if 0
   /* FIXME: Use the new journal API for detecting whether we’re writing to the
    * journal. See: https://github.com/systemd/systemd/issues/2473
    */
@@ -2016,7 +2016,7 @@ g_log_writer_format_fields (GLogLevelFla
   return g_string_free (gstring, FALSE);
 }
 
-#ifdef __linux__
+#if 0
 static int
 journal_sendv (struct iovec *iov,
                gsize         iovlen)
@@ -2064,6 +2064,10 @@ retry:
   if ((buf_fd = mkostemp (path, O_CLOEXEC|O_RDWR)) < 0)
     return -1;
 
+#ifndef HAVE_O_CLOEXEC
+  fcntl (buf_fd, F_SETFD, FD_CLOEXEC);
+#endif
+
   if (unlink (path) < 0)
     {
       close (buf_fd);
@@ -2130,7 +2134,7 @@ g_log_writer_journald (GLogLevelFlags   
                        gsize            n_fields,
                        gpointer         user_data)
 {
-#ifdef __linux__
+#if 0
   const char equals = '=';
   const char newline = '\n';
   gsize i, k;
