--- ./CMakeLists.txt
+++ ./CMakeLists.txt
@@ -298,6 +298,7 @@ set(TURBOJPEG_SO_VERSION 0.2.0)
 ###############################################################################
 
 if(MSVC)
+  set(CMAKE_EXE_LINKER_FLAGS "/MANIFEST:NO ${CMAKE_EXE_LINKER_FLAGS}")
   option(WITH_CRT_DLL
     "Link all ${CMAKE_PROJECT_NAME} libraries and executables with the C run-time DLL (msvcr*.dll) instead of the static C run-time library (libcmt*.lib.)  The default is to use the C run-time DLL only with the libraries and executables that need it."
     FALSE)
