--- ./libmariadb/CMakeLists.txt
+++ ./libmariadb/CMakeLists.txt
@@ -450,7 +450,7 @@ INSTALL(TARGETS libmariadb
 
 IF(WIN32)
   IF(${CMAKE_BUILD_TYPE} STREQUAL "RelWithDebInfo")
-    SET(PDB_LOCATION ${CC_BINARY_DIR}/libmariadb/RelWIthDebInfo/libmariadb.pdb)
+    SET(PDB_LOCATION ${CC_BINARY_DIR}/libmariadb/libmariadb.pdb)
    INSTALL(FILES ${PDB_LOCATION} DESTINATION "${INSTALL_LIBDIR}"
            COMPONENT Development)
   ENDIF()
