--- ./mDNSShared/dnssd_clientstub.c
+++ ./mDNSShared/dnssd_clientstub.c
@@ -578,8 +578,12 @@ static DNSServiceErrorType ConnectToServ
             }
             else 
             {
+                char *path = NULL;
+#ifndef USE_TCP_LOOPBACK
+                path = uds_serverpath;
+#endif
                 syslog(LOG_WARNING, "dnssd_clientstub ConnectToServer: connect() failed path:%s Socket:%d Err:%d Errno:%d %s", 
-                       uds_serverpath, sdr->sockfd, err, dnssd_errno, dnssd_strerror(dnssd_errno));
+                       path, sdr->sockfd, err, dnssd_errno, dnssd_strerror(dnssd_errno));
                 dnssd_close(sdr->sockfd); 
                 FreeDNSServiceOp(sdr); 
                 return kDNSServiceErr_ServiceNotRunning; 
