--- ./qtbase/src/plugins/platforms/windows/qwindowsservices.cpp
+++ ./qtbase/src/plugins/platforms/windows/qwindowsservices.cpp
@@ -53,13 +53,15 @@ static inline bool shellExecute(const QU
 #ifndef Q_OS_WINCE
     const QString nativeFilePath =
             url.isLocalFile() ? QDir::toNativeSeparators(url.toLocalFile()) : url.toString(QUrl::FullyEncoded);
-    const quintptr result =
-        reinterpret_cast<quintptr>(ShellExecute(0, 0,
-                                                reinterpret_cast<const wchar_t *>(nativeFilePath.utf16()),
-                                                0, 0, SW_SHOWNORMAL));
-    // ShellExecute returns a value greater than 32 if successful
-    if (result <= 32) {
-        qWarning("ShellExecute '%s' failed (error %s).", qPrintable(url.toString()), qPrintable(QString::number(result)));
+    SHELLEXECUTEINFO sei;
+    memset(&sei, 0, sizeof(sei));
+    sei.cbSize = sizeof(sei);
+    sei.lpFile = (wchar_t*)nativeFilePath.utf16();
+    sei.nShow = SW_SHOWNORMAL;
+
+    if (!ShellExecuteEx(&sei)) {
+        quintptr error = (quintptr) GetLastError();
+        qWarning("ShellExecuteEx '%s' failed (error 0x%x).", qPrintable(url.toString()), error);
         return false;
     }
     return true;
