--- ./makefile.msc
+++ ./makefile.msc
@@ -24,6 +24,9 @@ bzip2: lib
 lib: $(OBJS)
 	lib /out:libbz2.lib $(OBJS)
 
+bzip2.dll: $(OBJS)
+	link /def:libbz2.def /dll /out:bzip2.dll /implib:libbz2.lib $(OBJS)
+
 test: bzip2
 	type words1
 	.\\bzip2 -1  < sample1.ref > sample1.rb2
--- ./libbz2.def
+++ ./libbz2.def
@@ -1,4 +1,4 @@
-LIBRARY			LIBBZ2
+LIBRARY			BZIP2
 DESCRIPTION		"libbzip2: library for data compression"
 EXPORTS
 	BZ2_bzCompressInit
