--- ./src/libFLAC/cpu.c
+++ ./src/libFLAC/cpu.c
@@ -475,7 +475,7 @@ FLAC__uint32 FLAC__cpu_xgetbv_x86(void)
 {
 #if (defined _MSC_VER || defined __INTEL_COMPILER) && defined FLAC__AVX_SUPPORTED
 	return (FLAC__uint32)_xgetbv(0);
-#elif defined __GNUC__
+#elif __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 2) // Exclude OS X's gcc-4.2
 	FLAC__uint32 lo, hi;
 	asm volatile (".byte 0x0f, 0x01, 0xd0" : "=a"(lo), "=d"(hi) : "c" (0));
 	return lo;
